### DISCUSSION TITLE
Roland PD-5 triggering

### DISCUSSION THREAD

**Discussion by thijstriemstra** [corrados/edrumulus#36] [General]
I'm testing multiple Roland PD-5 single triggering pads with an ESP32 and this config:

| Pad | Type | MIDI note | GPIO |
| --- | --- | --- | --- |
| Roland PD-5 | Tom 1 | 49 | 34 |
| Roland PD-5 | Tom 2 | 48 | 39 |

and hitting the pad only produces midi notes when hitting the pad quite hard. The rim is very responsive though. I'm using the `thijstriemstra_prototype` branch.

I know the pads are old and much less responsive than mesh head pads but just wanted to check if this is it or if it can be improved.

Here's a video (with snare sound so it's better to hear vs toms):

https://user-images.githubusercontent.com/305679/148444022-4c99ffd5-47c1-45c6-a6fe-7c1ac75aa72c.mp4

**Comment by corrados** [comment]
That is very strange behavior. I have a PD8 which should be similar to your pad and it works very well with Edrumulus. Don't you have more than one of these pads? Are they behave all the same?

The main piezo should be located in the middle of the pad. If you record that piezo in a wav file and look at the signal in your wave recording software, does the level of the peaks go down if you strike the pad near the middle?

**Reply by thijstriemstra** [reply]
> Don't you have more than one of these pads? Are they behave all the same?

Yes, I bought 4 of them, second hand, for something like 30 euros, so I didn't have high hopes they were still any good after someone else probably played on them for years.

I will test all of them (on same tom2 circuit) and add .wav files for each. If I should do the same on my tom1 circuit, let me know, and I'll check that one for all 4 as well.

And my offer to send you one as a gift still stands!

**Reply by thijstriemstra** [reply]
The zip below contains 4 mono samples for each pad (A, B, C, D):

- low
- mid
- high
- clockwise (from top center of rim, clockwise)

Each sample has hits from left to right to left, then top to bottom to top (except for clockwise one of course); a cross basically.

[roland-pd5-toms.zip](https://github.com/corrados/edrumulus/files/7833676/roland-pd5-toms.zip)

**Reply by corrados** [reply]
Thanks for the files. What have you found out trying out the different pads on your ESP32? Do they all behave the same?

As in this [youtube video](https://www.youtube.com/watch?v=oVGX9X22DiI) (fast forward to minute 8:18), the piezo can get loose. Maybe that happened in your pad.

**Reply by corrados** [reply]
I added your signal to the Git repo: https://github.com/corrados/edrumulus/commit/ac3572c0138ce23a0e31fab576838f108f1c2320. Thank you for the recording.

**Reply by thijstriemstra** [reply]
thanks @corrados 

My original prototype config for the PD-5 tom1 and tom2 looks like:

```c++
 edrumulus.set_midi_notes       ( 4, 49, 55 ); // tom 1
 edrumulus.set_midi_notes       ( 5, 48, 50 ); // tom 2
```

but since there's no dual triggering, I'm not sure why it was defined like that so I changed it to:

```c++
edrumulus.set_midi_notes       ( 4, 49, 49 ); // tom 1
edrumulus.set_midi_notes       ( 5, 48, 48 ); // tom 2
```

**Reply by corrados** [reply]
If you do not have dual triggering, the second MIDI not is ignored.
